cmd/juju/commands/ssh.go

```diff
@@ -102,24 +112,48 @@ func (c *sshCommand) Info() *cmd.Info {
 	})
 }
 
-func (c *sshCommand) Init(args []string) error {
+func (c *sshCommand) Init(args []string) (err error) {
 	if len(args) == 0 {
 		return errors.Errorf("no target name specified")
 	}
-	c.Target, c.Args = args[0], args[1:]
+	c.modelType, err = c.ModelType()
+	if err != nil {
+		return err
+	}
+	if c.modelType == model.CAAS {
+		c.provider = &c.sshContainer
+	} else {
+		c.provider = &c.SSHCommon
+	}
+	c.provider.SetTarget(args[0])
+	c.provider.SetArgs(args[1:])
+	c.provider.setHostChecker(c.hostChecker)
 	return nil
 }
 
+type sshProvider interface {
```

Comment:
Maybe a comment here explaining that this interface is implemented by either either  a cass or iaas model instance